src/Symfony/Component/Notifier/Bridge/Gitter/Tests/GitterTransportFactoryTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Gitter\Tests;
+
+use Symfony\Component\Notifier\Bridge\Gitter\GitterTransportFactory;
+use Symfony\Component\Notifier\Tests\TransportFactoryTestCase;
+use Symfony\Component\Notifier\Transport\TransportFactoryInterface;
+
+/**
+ * @author Christin Gruber <c.gruber@touchdesign.de>
+ */
+final class GitterTransportFactoryTest extends TransportFactoryTestCase
+{
+    public function createFactory(): TransportFactoryInterface
+    {
+        return new GitterTransportFactory();
+    }
+
+    public function createProvider(): iterable
+    {
+        yield [
+            'gitter://api.gitter.im?room=testRoom',
+            'gitter://token@api.gitter.im?room=testRoom',
+        ];
+
+        yield [
+            'gitter://api.gitter.im/sub?room=testRoom',
+            'gitter://token@api.gitter.im/sub?room=testRoom',
+        ];
+
+        yield [
+            'gitter://api.gitter.im?room=testRoom',
+            'gitter://token@api.gitter.im/?room=testRoom',
+        ];
+
+        yield [
+            'gitter://api.gitter.im/sub?room=testRoom',
+            'gitter://token@api.gitter.im/sub?room=testRoom',
+        ];
+
+        yield [
+            'gitter://api.gitter.im/sub/sub-2?room=testRoom',
+            'gitter://token@api.gitter.im/sub/sub-2?room=testRoom',
+        ];
+    }
+
+    public function supportsProvider(): iterable
+    {
+        yield [true, 'gitter://token@host?room=testRoom'];
+        yield [false, 'invalidProvider://token@host?room=testRoom'];
```

Comment:
```suggestion
        yield [false, 'somethingElse://token@host?room=testRoom'];
```
This is what we use in other transports